students/SLammertink/Lesson03/strformat_lab.py

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env python3
+# UW Self paced Lesson 03 string lambda
+
+# Task 1
+''' Write a format string that will take the following four element tuple:
+( 2, 123.4567, 10000, 12345.67)
+and produce:
+'file_002 :   123.46, 1.00e+04, 1.23e+04' '''
+
+StrList = (2, 123.4567, 10000, 12345.67)
+
+def Task1():
+    print(f"file_{StrList[0]:03d} :  {StrList[1]:.2f}, {StrList[2]:.2e}, {StrList[3]:.2e}")
+
+
+# task 2
+
+'''Using your results from Task One, repeat the exercise, but this time using
+an alternate type of format string (hint: think about alternative ways to use .format()
+(keywords anyone?), and also consider f-strings if you’ve not used them already).
+'''
+
+
+def Task2():
+    print("file_{0:03d} :  {1:.2f}, {2:.2e}, {3:.2e}".format(StrList[0], StrList[1], StrList[2], StrList[3]))
+    print("file_%03d :  %.2f, %.2e, %.2e" % StrList)
+
+# Task 3
+
+''' Rewrite: "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3) to take an arbitrary number of values.'''
+
+def formatter(in_tuple):
+    formatter_string = "the " + str(len(in_tuple)) + " numbers are " + ", ".join(["{}"]*len(in_tuple)).format(*in_tuple)
+    print(formatter_string)
+
+# Task 4
+
+'''Given a 5 element tuple: ( 4, 30, 2017, 2, 27)
+use string formating to print: '02 27 2017 04 30' '''
+
+five_tuple = ( 4, 30, 2017, 2, 27)
+
+def TaskFour(five_tuple):
+    reformat = "{0:02d} {1:d} {2:d} {3:02d} {4:d}"
+    print(reformat.format(five_tuple[3], five_tu
+                           uit_list = ['oranges', 1.3, 'lemons', 1.1]
```

Comment:
Something odd going on here - typo perhaps?  Doesn't process correctly - five_tu?  uit_list?

Reply:
Hi must be a typo, that uit_list comes from a  different exercise..strange..it should be:
print(reformat.format(five_tuple[3], five_tuple[4], five_tuple[2], five_tuple[0], five_tuple[1]))